lib/bolt/util.rb

```diff
@@ -169,12 +169,20 @@ def postwalk_vals(data, skip_top = false, &block)
       # references to the same object and prevents endless recursion.
       # Credit to Jan Molic via https://github.com/rubyworks/facets/blob/master/LICENSE.txt
       def deep_clone(obj, cloned = {})
-        cloned[obj.object_id] if cloned.include?(obj.object_id)
+        return cloned[obj.object_id] if cloned.include?(obj.object_id)
 
         begin
           cl = obj.clone
-        rescue TypeError
-          # unclonable (TrueClass, Fixnum, ...)
+        rescue StandardError => e
```

Comment:
A common pattern for this is to use an array of error classes and splat them into the rescue statement.

```rb
error_types = [TypeError]
if defined?(Java::JavaLang::CloneNotSupportedException)
 error_types << Java::JavaLang::CloneNotSupportedException
end

begin
# ...
rescue *error_types
  # ...
end
```